source/agora/common/crypto/Key.d

```diff
@@ -64,10 +66,15 @@ public struct KeyPair
     /// Create a keypair from a `Seed`
     public static KeyPair fromSeed (Seed seed) nothrow @nogc
     {
-        SecretKey sk;
+        BitBlob!(crypto_sign_ed25519_SECRETKEYBYTES * 8) sk_data;
         PublicKey pk;
-        if (crypto_sign_seed_keypair(pk[].ptr, sk[].ptr, seed[].ptr) != 0)
+        if (crypto_sign_seed_keypair(pk[].ptr, sk_data[].ptr, seed[].ptr) != 0)
             assert(0);
+
+        Scalar x25519_sk;
+        if (crypto_sign_ed25519_sk_to_curve25519(cast(ubyte*)(x25519_sk[].ptr), sk_data[].ptr) != 0)
+            assert(0);
+        SecretKey sk = SecretKey(x25519_sk[]);
```

Comment:
Are those different ? I expect they just hold the same value, don't they ?

Reply:
No, they are not. The data of the `SecretKey` has the `seed` and `Point` in its 64 bytes storage as follows.
https://github.com/jedisct1/libsodium/blob/57d950a54e6f7743084092ba5d31b8fa0641eab2/src/libsodium/crypto_sign/ed25519/ref10/keypair.c#L26-L27